tests/test_integration.py

```diff
@@ -150,6 +180,34 @@ def test_worker_service_called_multiple_times(self):
         eq_(payload['backend'], {'app': 'django', 'app_type': 'web'})
         eq_(payload['database'], {'app': 'postgres', 'app_type': 'db'})
 
+    def test_worker_http_error_logging(self):
+        # Tests the logging http error logic
+        tracer = self.tracer
+        self.tracer.writer.api = FlawedAPI(Tracer.DEFAULT_HOSTNAME, Tracer.DEFAULT_PORT)
+        tracer.trace('client.testing').finish()
+        tracer.trace('client.testing').finish()
+        eq_(tracer.writer._worker._last_error_ts, 0)
```

Comment:
this check could be flaky because you're relying that this thread is faster than the API worker